helix-term/src/ui/picker.rs

```diff
@@ -113,17 +146,35 @@ impl<T: 'static> Component for FilePicker<T> {
             horizontal: 1,
         };
         let inner = inner.inner(&margin);
-
         block.render(preview_area, surface);
 
-        if let Some((doc, line)) = self.current_file(cx.editor).and_then(|(path, range)| {
-            cx.editor
-                .document_by_path(&path)
-                .or_else(|| self.preview_cache.get(&path))
-                .zip(Some(range))
-        }) {
+        if let Some((path, range)) = self.current_file(cx.editor) {
+            let cache = self.preview_cache.get(&path);
+            let doc = match cache {
+                Some(Preview::Binary | Preview::LargeFile | Preview::NotFound) => {
+                    surface.set_stringn(
+                        inner.x,
+                        inner.y,
+                        match cache {
+                            Some(Preview::Binary) => "<Binary file>",
+                            Some(Preview::LargeFile) => "<File too large to preview>",
+                            Some(Preview::NotFound) => "<File not found>>",
+                            _ => unreachable!(),
+                        },
+                        inner.width as usize,
+                        text,
+                    );
+                    return;
+                }
+                Some(Preview::Document(doc)) => doc,
+                None => match cx.editor.document_by_path(&path) {
+                    Some(doc) => doc,
+                    None => return,
+                },
```

Comment:
This can also be moved to `get_preview()` and we can purely match on `Preview` variants.